### ISSUE TITLE
Compilation failure with GCC 6

### ISSUE THREAD

**Issue by verdurin** [eturro/mmseq#9]
There is a build failure on Fedora Rawhide, which is using GCC 6.

Here is the error:

```
hitsio.cpp: In member function 'bool HitsfileReader::readReadMapRecordTranscriptIDSchema0(std::__cxx11::string&)':
hitsio.cpp:346:35: error: cannot convert 'std::basic_istream<char>' to 'bool' in return
  return(getline(ifs, transcriptID));
                                   ^
```

The person who first spotted this suggested the following fix:

```
return(static_cast<bool>(getline(ifs, transcriptID)));
```

**Comment by verdurin** [comment]
Actually there's another error:

```
g++ -DVERSION=1.0.8 -DARMA_64BIT_WORD -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -fopenmp -O3 uh.o sokal.o mmcollapse.cpp -o ../bin/mmcollapse -lgsl -lgslcblas -lboost_regex -lbam -lboost_iostreams -llapack -lblas -lgfortran -Wl,--as-needed -lz -larmadillo
In file included from bms.cpp:21:0:
bms.hpp:138:25: error: 'constexpr' needed for in-class initialization of static data member 'const double BMS::g' of non-integral type [-fpermissive]
   static const double g=2.0;
                         ^~~
```

**Comment by verdurin** [comment]
I've addressed these problems in #10

**Comment by eturro** [comment]
Much appreciated!
